File: API/Domain/Asset/AssetSupplySource.cs
using API.Domain.Shared;

namespace API.Domain.Asset;

public class AssetSupplySource : TenantEntity
{
    public required string AssetId { get; set; }
    public Asset? Asset { get; set; }
    
    public required string SupplierId { get; set; }
    public AssetSupplier? Supplier { get; set; }

    public required string SupplierReference { get; set; }
    public string? QuantityUnit { get; set; }

    public Dictionary<float, double> Prices { get; set; } = [];
}

File: client/src/core/data/entities/asset/asset-supply-source.ts
import { BaseEntity } from '../base-entity.ts';

export interface AssetSupplySource extends BaseEntity {
  supplierId: string;
  supplierName?: string;

  supplierReference: string;
  quantityUnit?: string;
  prices: Record<number, number>;
}

export interface AssetSupplySourceForm {
  id?: string;
  assetId: string;
  supplierId: string;
  supplierReference: string;
  quantityUnit?: string;
  prices: Record<number, number>;
}